src/qt/bitcoin.cpp

```diff
@@ -654,9 +655,23 @@ int GuiMain(int argc, char* argv[])
     initTranslations(qtTranslatorBase, qtTranslator, translatorBase, translator);
 
     g_software_expiry = gArgs.GetIntArg("-softwareexpiry", DEFAULT_SOFTWARE_EXPIRY);
-    if (IsThisSoftwareExpired(GetTime())) {
-        QMessageBox::critical(nullptr, QObject::tr("Software expired"), QObject::tr("This software is expired, and may be out of consensus. You must choose to upgrade or override this expiration."));
-        return EXIT_FAILURE;
+    if (IsThisSoftwareExpiringSoon(GetTime())) {
+        QMessageBox::warning(nullptr, QObject::tr("Software expires soon"), QObject::tr("This software expires soon, and may fall out of consensus. You must choose to upgrade, or override this expiration with the 'softwareexpiry' option (Unix datetime, or 0 for no expiry)."));
+    } else if (IsThisSoftwareExpired(GetTime())) {
+        bool override_expiry_ok;
+        QString override_expiry = QInputDialog::getText(
+            nullptr,
+            QObject::tr("Software Expired"),
+            QObject::tr("This software is expired, and may be out of consensus.\nYou must choose to upgrade, or override this expiration by typing:\nI accept this software may be unsafe."),
+            QLineEdit::Normal,
+            "",
+            &override_expiry_ok);
+        if (!override_expiry_ok || override_expiry != QObject::tr("I accept this software may be unsafe.")) {
+            return EXIT_FAILURE;
+        }
+
+        gArgs.ModifyRWConfigFile("softwareexpiry", "0");
```

Comment:
Probably shouldn't persist forever, and should reset to default if the user upgrades?